eth_tester/main.py

```diff
@@ -333,9 +333,17 @@ def get_transaction_receipt(self, transaction_hash):
         raw_transaction_hash = self.normalizer.normalize_inbound_transaction_hash(
             transaction_hash,
         )
+        transaction = self.get_transaction_by_hash(transaction_hash)
         raw_receipt = self.backend.get_transaction_receipt(raw_transaction_hash)
         self.validator.validate_outbound_receipt(raw_receipt)
         receipt = self.normalizer.normalize_outbound_receipt(raw_receipt)
+
+        byzantium_fork_block = None
+        if 'FORK_BYZANTIUM' in self.get_supported_forks():
+            byzantium_fork_block = self.get_fork_block('FORK_BYZANTIUM')
+        status = int.from_bytes(receipt.pop('state_root'), byteorder='big')
+        if byzantium_fork_block is not None and byzantium_fork_block <= transaction['block_number']:
+            receipt['status'] = status
```

Comment:
Instead of mutating this value, we generally have chosen to use non-mutative approach like:

```
from cytoolz import assoc

return assoc(receipt, 'status', status)
```

This generates a new dictionary rather than mutating the existing one.